src/libs/sfdk/vboxvirtualmachine.cpp

```diff
@@ -0,0 +1,1008 @@
+        return runner_;
+    };
+
+    CommandQueue::BatchId batch = commandQueue()->beginBatch();
+    enqueue(rargs, batch);
+    enqueue(aargs, batch);
+    enqueue(sargs, batch);
+    commandQueue()->enqueueCheckPoint(context, std::bind(functor, true));
+    commandQueue()->endBatch();
+}
+
+void VBoxVirtualMachinePrivate::doAddPortForwarding(const QString &ruleName,
+        const QString &protocol, quint16 hostPort, quint16 emulatorVmPort,
+        const QObject *context, const Functor<bool> &functor)
+{
+    Q_Q(VBoxVirtualMachine);
+    Q_ASSERT(context);
+    Q_ASSERT(functor);
+
+    doRemovePortForwarding(ruleName, Sdk::instance(), [](bool) {/* noop */});
+
+    qCDebug(vms) << "Setting port forwarding for" << q->uri().toString() << "from"
+        << hostPort << "to" << emulatorVmPort;
+
+    QStringList arguments;
+    arguments.append(QLatin1String(MODIFYVM));
+    arguments.append(q->name());
+    arguments.append(QLatin1String(NATPF1));
+    arguments.append(QString::fromLatin1("%1,%2,,%3,,%4")
+                     .arg(ruleName).arg(protocol).arg(hostPort).arg(emulatorVmPort));
+
+    auto runner = std::make_unique<VBoxManageRunner>(arguments);
+    QObject::connect(runner.get(), &VBoxManageRunner::done, context, functor);
+    commandQueue()->enqueueImmediate(std::move(runner));
+}
+
+void VBoxVirtualMachinePrivate::doRemovePortForwarding(const QString &ruleName,
+        const QObject *context, const Functor<bool> &functor)
+{
+    Q_Q(VBoxVirtualMachine);
+    Q_ASSERT(context);
+    Q_ASSERT(functor);
+
+    qCDebug(vms) << "Deleting port forwarding rule" << ruleName << "from" << q->uri().toString();
+
+    QStringList arguments;
+    arguments.append(QLatin1String(MODIFYVM));
+    arguments.append(q->name());
+    arguments.append(QLatin1String(NATPF1));
+    arguments.append(QLatin1String(DELETE));
+    arguments.append(ruleName);
+
+    auto runner = std::make_unique<VBoxManageRunner>(arguments);
+    QObject::connect(runner.get(), &VBoxManageRunner::done, context, functor);
+    commandQueue()->enqueue(std::move(runner));
+}
+
+void VBoxVirtualMachinePrivate::doSetReservedPortForwarding(ReservedPort which, quint16 port,
+        const QObject *context, const Functor<bool> &functor)
+{
+    Q_Q(VBoxVirtualMachine);
+    Q_ASSERT(context);
+    Q_ASSERT(functor);
+
+    qCDebug(vms) << "Setting reserved port forwarding" << which << "for" << q->uri().toString()
+        << "to" << port;
+
+    QString ruleName;
+    int guestPort;
+    switch (which) {
+        case VirtualMachinePrivate::SshPort:
+            ruleName = QLatin1String(SSH_NATPF_RULE_NAME);
+            guestPort = 22;
+            break;
+        case VirtualMachinePrivate::WwwPort:
+            ruleName = QLatin1String(WWW_NATPF_RULE_NAME);
+            guestPort = 8080;
+            break;
+    }
+    const QString rule = QString::fromLatin1(NATPF_RULE_TEMPLATE)
+        .arg(ruleName).arg(port).arg(guestPort);
```

Comment:
Now that you added -Werror, this no longer compiles. error: ‘guestPort’ may be used uninitialized in this function.

Reply:
Fixed